src/pybind/mgr/cephadm/migrations.py

```diff
@@ -126,3 +130,30 @@ def convert_to_explicit(spec: ServiceSpec) -> None:
             convert_to_explicit(spec)
 
         return True
+
+    def migrate_1_2(self) -> bool:
+        """
+        After 15.2.4, we unified some service IDs: MONs, MGRs etc no longer have a service id.
+        Which means, the service names changed:
+
+        mon.foo -> mon
+        mgr.foo -> mgr
+
+        This fixes the data structure consistency
+        """
+        bad_specs = {}
+        for name, spec in self.mgr.spec_store.specs.items():
+            if name != spec.service_name():
+                bad_specs[name] = (spec.service_name(), spec)
+
+        for old, (new, old_spec) in bad_specs.items():
+            if new not in self.mgr.spec_store.specs:
+                old_spec.unmanaged = True
+                self.mgr.spec_store.save(old_spec)
+                self.mgr.spec_store.rm(old)
+            else:
+                new_spec = self.mgr.spec_store.specs[new]
+                new_spec.unmanaged = True
+                self.mgr.spec_store.save(new_spec)
+                self.mgr.spec_store.rm(old)
```

Comment:
Just a suggestion to easier spot the difference between both cases.
```suggestion
            spec = old_spec if new not in self.mgr.spec_store.specs else self.mgr.spec_store.specs[new]
            spec.unmanaged = True
            self.mgr.spec_store.save(spec)
            self.mgr.spec_store.rm(old)
```